vcd/resource_vcd_vapp.go

```diff
@@ -152,7 +152,7 @@ func resourceVcdVAppCreate(ctx context.Context, d *schema.ResourceData, meta int
 
 	d.SetId(vapp.VApp.ID)
 
-	return resourceVcdVAppUpdate(ctx, d, meta)
+	return resourceVcdVAppRead(ctx, d, meta)
```

Comment:
Note to reviewers, I noticed this one was calling Update after Create. I think it should be a Read like any other resource, let me know otherwise.

Reply:
The call to Update is necessary, as otherwise it would not set the vApp lease.
We are trying not to do it in new resources, but this is how it was done at the time

Reply:
Thanks for the explanation. I have reverted the change.